FBTweak/FBTweak.m

```diff
@@ -180,13 +180,21 @@ @implementation FBPersistentTweak
 - (instancetype)initWithIdentifier:(NSString *)identifier name:(NSString *)name
                       defaultValue:(FBTweakValue)defaultValue {
   if (self = [super initWithIdentifier:identifier name:name defaultValue:defaultValue]) {
+    id currentValue;
     NSData *archivedValue = [[NSUserDefaults standardUserDefaults] objectForKey:identifier];
     if (archivedValue != nil && [archivedValue isKindOfClass:[NSData class]]) {
-      self.currentValue =
+      currentValue =
           [NSKeyedUnarchiver unarchivedObjectOfClasses:[FBPersistentTweak validValueClasses]
                                               fromData:archivedValue error:nil];
     } else {
-      self.currentValue = archivedValue;
+      currentValue = archivedValue;
+    }
+
+    // Using \c class instead of \c classForKeyedArchiver might lead to unexpected behaviour because
+    // defaultValue might be a hidden class like \c NSTaggedPointerString, in which case the check
+    // will fail. \c classForKeyedArchiver will return the class we want to check against.
+    if ([currentValue isKindOfClass:[self.defaultValue classForKeyedArchiver]]) {
+      self.currentValue = currentValue;
```

Comment:
do we want to force `self.currentValue = nil` so it'll be persistent?

Reply:
From outside the behaviour is the same, so not sure if we want to waste time on this.